opensrp-reveal/src/main/java/org/smartregister/reveal/view/OfflineMapsActivity.java

```diff
@@ -0,0 +1,136 @@
+
+import io.ona.kujaku.downloaders.MapBoxOfflineResourcesDownloader;
+import io.ona.kujaku.utils.LogUtil;
+
+public class OfflineMapsActivity extends AppCompatActivity implements OfflineMapDownloadCallback {
+
+    private static final String TAG = OfflineMapsActivity.class.getName();
+
+    private ViewPagerAdapter adapter;
+
+    private static  final int AVAILABLE_OFFLINE_MAPS_FRAGMENT_INDEX = 0;
+    private static  final int DOWNLOADED_OFFLINE_MAPS_FRAGMENT_INDEX = 1;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        setContentView(R.layout.activity_offline_maps);
+
+        setupViews();
+
+        getOfflineDownloadedRegions();
+    }
+
+    protected void setupViews() {
+        TabLayout tabLayout = findViewById(R.id.tabs);
+        ViewPager viewPager = findViewById(R.id.viewpager);
+        tabLayout.setupWithViewPager(setupViewPager(viewPager));
+    }
+
+    protected ViewPager setupViewPager(ViewPager viewPager) {
+        adapter = new ViewPagerAdapter(getSupportFragmentManager());
+
+        AvailableOfflineMapsFragment availableOfflineMapsFragment = AvailableOfflineMapsFragment.newInstance(this.getIntent().getExtras());
+        availableOfflineMapsFragment.setOfflineMapDownloadCallback(this);
+        adapter.addFragment(availableOfflineMapsFragment, this.getString(R.string.available).toUpperCase());
+
+        DownloadedOfflineMapsFragment downloadedOfflineMapsFragment = DownloadedOfflineMapsFragment.newInstance(this.getIntent().getExtras());
+        downloadedOfflineMapsFragment.setOfflineMapDownloadCallback(this);
+        adapter.addFragment(downloadedOfflineMapsFragment, this.getString(R.string.downloaded).toUpperCase());
+
+        viewPager.setAdapter(adapter);
+
+        return viewPager;
+    }
+
+    @Override
+    public void onMapDownloaded(OfflineMapModel offlineMapModel) {
+        DownloadedOfflineMapsFragment downloadedOfflineMapsFragment = (DownloadedOfflineMapsFragment)  adapter.getItem(DOWNLOADED_OFFLINE_MAPS_FRAGMENT_INDEX);
+        downloadedOfflineMapsFragment.updateDownloadedMapsList(offlineMapModel);
+    }
+
+    @Override
+    public void onOfflineMapDeleted(OfflineMapModel offlineMapModel) {
+        AvailableOfflineMapsFragment availableOfflineMapsFragment = (AvailableOfflineMapsFragment)  adapter.getItem(AVAILABLE_OFFLINE_MAPS_FRAGMENT_INDEX);
+        availableOfflineMapsFragment.updateOperationalAreasToDownload(offlineMapModel);
+    }
+
+    private void getOfflineDownloadedRegions() {
+        Mapbox.getInstance(this, BuildConfig.MAPBOX_SDK_ACCESS_TOKEN);
+
+        OfflineManager offlineManager = OfflineManager.getInstance(this);
+        offlineManager.listOfflineRegions(new OfflineManager.ListOfflineRegionsCallback() {
+            @Override
+            public void onList(final OfflineRegion[] offlineRegions) {
+                List<String> offlineRegionNames = new ArrayList<>();;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [An empty statement (semicolon) not part of a loop](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=4217760)